src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1862,8 +1837,6 @@ void SystemDictionary::check_constraints(unsigned int name_hash,
       }
     }
 
-    DEBUG_ONLY(if (is_parallelCapable(class_loader)) verify_placeholder(name, loader_data));
```

Comment:
I can't help but feel that somewhere in the loading process there should be a more general assertion that, not only is there a placeholder present, but that it is for this thread. Though such a check would also need to check the kind of placeholder found. Future RFE perhaps.

Reply:
In this case, we call check_constraints() after loadClass so the placeholder doesn't exist for parallelCapable class loaders (we do want one for the define class path).  Not sure what sort of assertion would be helpful here.